tests/platform_helper/test_command_environment.py

```diff
@@ -329,28 +327,94 @@ class TestGenerate:
+        "environment_config, expected_vpc",
+        [
+            ({"test": {}}, None),
+            ({"test": {"vpc": "vpc1"}}, "vpc1"),
+            ({"*": {"vpc": "vpc2"}, "test": None}, "vpc2"),
+            ({"*": {"vpc": "vpc3"}, "test": {"vpc": "vpc4"}}, "vpc4"),
+        ],
+    )
     def test_generate(
-        self, mock_get_aws_session, mock_get_vpc_id, mock_get_subnet_ids, mock_get_cert_arn, fakefs
+        self,
+        mock_get_aws_session,
+        mock_get_vpc_id,
+        mock_get_subnet_ids,
+        mock_get_cert_arn,
+        fakefs,
+        is_terraform,
+        environment_config,
+        expected_vpc,
     ):
         from dbt_platform_helper.commands.environment import generate
 
+        default_conf = environment_config.get("*", {})
+        default_conf["accounts"] = {
+            "deploy": {"name": "non-prod-acc", "id": "1122334455"},
+            "dns": {"name": "non-prod-dns-acc", "id": "6677889900"},
+        }
+        environment_config["*"] = default_conf
+
         mocked_session = MagicMock()
         mock_get_aws_session.return_value = mocked_session
         fakefs.add_real_directory(
             BASE_DIR / "tests" / "platform_helper", read_only=False, target_path="copilot"
         )
+        fakefs.create_file(
+            PLATFORM_CONFIG_FILE,
+            contents=yaml.dump({"application": "my-app", "environments": environment_config}),
+        )
+        if is_terraform:
+            fakefs.create_dir("./terraform")
+
         result = CliRunner().invoke(generate, ["--name", "test"])
 
         actual = yaml.safe_load(Path("copilot/environments/test/manifest.yml").read_text())
         expected = yaml.safe_load(
             Path("copilot/fixtures/test_environment_manifest.yml").read_text()
         )
 
-        mock_get_vpc_id.assert_called_once_with(mocked_session, "test", None)
+        mock_get_vpc_id.assert_called_once_with(mocked_session, "test", expected_vpc)
         mock_get_subnet_ids.assert_called_once_with(mocked_session, "vpc-abc123")
         mock_get_cert_arn.assert_called_once_with(mocked_session, "test")
+
         assert actual == expected
         assert "File copilot/environments/test/manifest.yml created" in result.output
+        if is_terraform:
+            assert "File terraform/environments/test/main.tf created" in result.output
+        else:
+            assert "File terraform/environments/test/main.tf created" not in result.output
+
+    def test_fail_early_if_platform_config_invalid(self, fakefs):
+        from dbt_platform_helper.commands.environment import generate
+
+        fakefs.add_real_directory(
+            BASE_DIR / "tests" / "platform_helper", read_only=False, target_path="copilot"
+        )
+        fakefs.create_file(PLATFORM_CONFIG_FILE, contents=yaml.dump({}))
+
+        result = CliRunner().invoke(generate, ["--name", "test"])
+
+        assert result.exit_code != 0
+        assert "Missing key: 'application'" in result.output
```

Comment:
Is this the only thing that's actually considered invalid?

Or are other invalinesses handled elsewhere?

Reply:
It's the only thing we care about in this test, yes. I wanted to move the config validation earlier in the command so that the command fails early if the config is invalid. So a missing key achieves that. test_validation.py is where the comprehensive config validation takes place.